pulumi/infra/policies.py

```diff
@@ -0,0 +1,156 @@
+                            "ec2messages:GetMessages",
+                            "ec2messages:SendReply",
+                        ],
+                        "Resource": "*",
+                    },
+                ],
+            }
+        ),
+        opts=opts,
+    )
+    return aws.iam.RolePolicyAttachment(
+        f"attach_{policy_name}",
+        role=role.name,
+        policy_arn=policy.arn,
+        opts=opts,
+    )
+
+
+def _attach_CloudWatchAgentServerPolicy(
+    role: aws.iam.Role, opts: pulumi.ResourceOptions
+) -> aws.iam.RolePolicyAttachment:
+    """
+    https://github.com/grapl-security/issue-tracker/issues/106
+    Moving away from managed server policies.
+    """
+    policy_name = "demanaged_CloudWatchAgentServerPolicy"
+    policy = aws.iam.Policy(
+        policy_name,
+        policy=json.dumps(
+            {
+                "Version": "2012-10-17",
+                "Statement": [
+                    {
+                        "Effect": "Allow",
+                        "Action": [
+                            "cloudwatch:PutMetricData",
+                            "ec2:DescribeVolumes",
+                            "ec2:DescribeTags",
+                            "logs:PutLogEvents",
+                            "logs:DescribeLogStreams",
+                            "logs:DescribeLogGroups",
+                            "logs:CreateLogStream",
+                            "logs:CreateLogGroup",
+                        ],
+                        "Resource": "*",
+                    },
+                    {
+                        "Effect": "Allow",
+                        "Action": ["ssm:GetParameter"],
+                        "Resource": "arn:aws:ssm:*:*:parameter/AmazonCloudWatch-*",
+                    },
+                ],
+            }
+        ),
+        opts=opts,
+    )
+    return aws.iam.RolePolicyAttachment(
+        f"attach_{policy_name}",
+        role=role.name,
+        policy_arn=policy.arn,
+        opts=opts,
+    )
+
+
+def _attach_policy_to_ship_logs_to_cloudwatch(
+    role: aws.iam.Role, log_group: aws.cloudwatch.LogGroup, opts: pulumi.ResourceOptions
+) -> aws.iam.RolePolicyAttachment:
+    policy_name = "policy_to_ship_logs_to_cloudwatch"
+    policy = log_group.arn.apply(
+        lambda arn: aws.iam.Policy(
+            policy_name,
+            policy=json.dumps(
+                {
+                    "Version": "2012-10-17",
+                    "Statement": [
+                        {
+                            "Effect": "Allow",
+                            "Action": [
+                                "logs:CreateLogGroup",
+                                "logs:CreateLogStream",
+                                "logs:PutLogEvents",
+                                "logs:DescribeLogStreams",
+                            ],
+                            "Resource": f"{arn}:*",
+                        }
+                    ],
```

Comment:
Looking closer, this policy appears to be a strict subset of the managed policy created above, and we attach it to the same Role. I don't think this is actually needed.